utils/cdi-provider/src/main/java/io/smallrye/config/cdi/providers/Name.java

```diff
@@ -0,0 +1,22 @@
+package io.smallrye.config.cdi.providers;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import javax.enterprise.util.Nonbinding;
+import javax.inject.Qualifier;
+
+/**
+ * The define the name of a config source
+ * 
+ * @author <a href="mailto:phillip.kruger@redhat.com">Phillip Kruger</a>
+ */
+@Qualifier
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ ElementType.METHOD, ElementType.FIELD })
+public @interface Name {
```

Comment:
Does it make sense to re-use `@Named` instead?

Reply:
It does, but I could not get it right :(

If anyone can help me on how to do that, that would be awesome. The problem I have is that config sources that get @Produces is dynamic. So this works with @Named:

    ```
@Produces
    @Named("SysPropConfigSource")
    public ConfigSource produceConfigSource() {
        return configSourceMap.get("SysPropConfigSource");
    }
```
(So I do an explicit @Produces of a known Config Source)

However, if I look through all config sources, I can not get @Named to work:

```
        @Produces
        @Named
        public ConfigSource produceConfigSource(final InjectionPoint injectionPoint) {
            Set<Annotation> qualifiers = injectionPoint.getQualifiers();
            String name = getName(qualifiers);
            return configSourceMap.get(name);
        } 
```
That is why I created my own Qualifier. but would love to reuse @Named. 
Any syuggestions ?

Reply:
@phillip-kruger Your `@Name` has nonbinding value, that means actual value on the `String` will be disregarded for resolution.
That's not the case with `@Named` where this is binding. E.g. if you have an injection points `@Inject @Name("foo") ConfigSource foo1` and `@Inject @Name("bar") ConfigSource bar1` then those can be served by the same producer method which has annotation `@Produces @Name("something")` because the annotation value is not taken into account.
With `@Named` you would need producer for each specific String value. I suppose this is the culprit here, does that make sense?

Reply:
Yes it does ! Thanks ! I'll add some test to make sure this works as expected.

Reply:
Added some test. It's working as expected, can not use @Named because of above mentioned binding.